src/utils/network-monitor.ts

```
@@ -1038,6 +1074,42 @@ export class NetworkMonitor {
     }
   }
 
+  async updateLastProcessedBlock(job: BlockJob): Promise<void> {
+    if (this.apiService === undefined) {
+      this.structuredLog(job.network, `updateLastProcessedBlock: API is undefined`)
+      return
+    }
+
+    let processType: BlockHeightProcessType | undefined
+
+    if (this.parent.constructor.name === 'Indexer') {
+      processType = BlockHeightProcessType.INDEXER
+    } else if (this.parent.constructor.name === 'Operator') {
+      processType = BlockHeightProcessType.OPERATOR
+    }
+
+    if (processType === undefined) {
```

Should this throw an error instead of just returning?